src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1341,12 +1341,14 @@ private Owner setupOwner(Principal principal, String ownerKey) {
         if (ownerKey == null && (principal instanceof UserPrincipal)) {
             // check for this cast?
             List<String> ownerKeys = ((UserPrincipal) principal).getOwnerKeys();
-
-            if (ownerKeys.size() != 1) {
+            if (ownerKeys.size() == 0 ||
+                (ownerKeys.size() > 1 &&
+                ((UserPrincipal) principal).getPrimaryOwner() == null)) {
```

Comment:
This can probably be merged/simplified. We don't need to do a getOwnerKeys. We only need to check if `getPrimaryOwner() == null` and throw the exception, since we consider having a single/primary owner the only valid state

Reply:
Actually, nevermind, it's probably better to do what Chris mentioned in his comment